TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -10,5 +16,136 @@ public CodeElement CreateCommitStatement(CodeElementsParser.CommitStatementConte
+        {
+            SyntaxProperty<SelectionModifier> selectionModifier = null;
+            if (context.SQL_ALL() != null)
+            {
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+            }
+            else if (context.SQL_DISTINCT() != null)
+            {
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+            }
+
+            if (context.star() != null)
+            {
+                return new SelectClause(selectionModifier, new StarSelection());
+            }
+
+            if (context.selections() != null)
+            {
+                var dotStarSelections = new List<DotStarSelection>();
+                int l = context.selections().ChildCount;
+                foreach (var selection in context.selections().selection())
+                {
+                    dotStarSelections.Add(CreateDotStarSelection(selection));
+                }
+
+                return new SelectClause(selectionModifier, dotStarSelections);
+            }
+
+            return null;
+        }
+
+        public DotStarSelection CreateDotStarSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.dotStarSelection() != null)
+            {
+                Token name = null;
+                Token schemaName = null;
+                Token dbms = null;
+                if ((context.dotStarSelection().tableOrViewOrCorrelationName().Name != null))
+                {
+                    name = context.dotStarSelection().tableOrViewOrCorrelationName().Name as Token;
+                    if (context.dotStarSelection().tableOrViewOrCorrelationName().SchemaName != null)
+                    {
+                        schemaName =
+                            context.dotStarSelection().tableOrViewOrCorrelationName().SchemaName as Token;
+                        if (context.dotStarSelection().tableOrViewOrCorrelationName().DBMS != null)
+                        {
+                            dbms =
+                                context.dotStarSelection().tableOrViewOrCorrelationName().DBMS as Token;
+                        }
+                    }
+                }
+                SymbolReference fullName = CreateSymbolReference(name, schemaName, dbms);
+                return new DotStarSelection(fullName);
+            }
+
+            return null;
+
+        }
+
+        private SymbolReference CreateSymbolReference(Token nameToken, Token qualifierToken, Token topLevelQualifierToken)
+        {
+            if (nameToken != null)
+            {
+                SymbolReference name = new SymbolReference(new AlphanumericValue(nameToken),
+                    SymbolType.SqlTableOrViewOrCorrelationName);
```

Comment:
I'm not sure this SymbolType is appropriate because it a _part_ of a table or view name. The `SqlTableOrViewOrCorrelationName`  suits for the whole name but not necessarily for its parts. So either we add multiple types like `SqlDbmsName`, `SqlSchemaName`, `SqlTableName`, `SqlColumnName`, or we use `SqlIdentifier` as a generic SymbolType. @smedilol, @mayanje what do you think ?